### DISCUSSION TITLE
Easiest way to check if volume contains another volume

### DISCUSSION THREAD

**Discussion by renatobellotti** [pyvista/pyvista#1414] [Q&A] [answered]
Hi,

I have two structures and am trying to determine if one is fully contained in the other. The volumes are given as point clouds. My approach would have been to take a 3D Delaunay triangulation, and then apply boolean operations to see if the intersection volume is equal to the smaller volumes. Apparently, this is not possible due to missing orientations. Does somebody know another way to solve the problem?

Best,
Renato

**Comment by akaszynski** [comment, marked as answer]
3D Delaunay triangulation should be possible even without orientations.  I'd try attempting 3d Delaunay triangulation first with [delaunay_3d](https://docs.pyvista.org/core/filters.html?highlight=delaunay#pyvista.DataSetFilters.delaunay_3d) and then use [select_enclosed_points](https://docs.pyvista.org/core/filters.html?highlight=inside#pyvista.DataSetFilters.select_enclosed_points).

**Reply by renatobellotti** [reply]
Thank you very much for your answer! Is there a way to get the overlap volume, too?

**Reply by akaszynski** [reply]
I'd use the Boolean filters.  We've improved them a lot in the main branch, you'll have to install it from the development version.

You'll probably want:

https://dev.pyvista.org/api/core/_autosummary/_autosummary/pyvista.PolyDataFilters.boolean_union.html

Install the latest development version with:

python -m pip install git+https://github.com/pyvista/pyvista.git

**Reply by renatobellotti** [reply]
Thanks a lot for your answer. I think I can solve my problem without the overlap volumes, but I will try your suggestion if the situation changes again.
